nip47/event_handler.go

```diff
@@ -93,12 +96,21 @@ func (svc *nip47Service) HandleEvent(ctx context.Context, relay nostrmodels.Rela
 		}
 	}
 
-	ss, err := nip04.ComputeSharedSecret(app.AppPubkey, appWalletPrivKey)
+	var ss []byte
+	var ck [32]byte
+	isNip04Encrypted := strings.Contains(event.Content, "?iv=")
+	if isNip04Encrypted {
+		ss, err = nip04.ComputeSharedSecret(app.AppPubkey, appWalletPrivKey)
+	} else {
+		ck, err = nip44.GenerateConversationKey(app.AppPubkey, appWalletPrivKey)
```

Comment:
I am wondering if we can better abstract the differences between NIP-04 and NIP-44, so the event handler does not need to know about it.

e.g. the event handler interacts with this abstraction and just passes the version number. Then the abstraction will either use NIP-04 or NIP-44, and all the conditionals will be isolated there.

And then we do not need to e.g. pass `ss` and `ck` separately, but have an encryption object that knows how to encrypt or decrypt based on the version it has?

Reply:
Yeah was thinking of the same, should we make a encrypt/decrypt function that takes in event (which has version and pubkey), privkey and content which conditionally does nip04/44?